core/container/externalbuilder/externalbuilder.go

```diff
@@ -268,10 +268,12 @@ type Builder struct {
 	Logger       *flogging.FabricLogger
 	Name         string
 	MSPID        string
+	signalChan   <-chan struct{}
+	doneChan     chan<- struct{}
 }
 
 // CreateBuilders will construct builders from the peer configuration.
-func CreateBuilders(builderConfs []peer.ExternalBuilder, mspid string) []*Builder {
+func CreateBuilders(builderConfs []peer.ExternalBuilder, mspid string, signalChan <-chan struct{}, doneChan chan<- struct{}) []*Builder {
```

Comment:
Instead of wiring these, and instead of wiring them to the constructor, can we wire a context through to the appropriate methods?

A context would allow us to orchestrate cancelation using a relatively common pattern.